pkg/scheduler/queue/multi_queuing_algorithm_tree_queue_benchmark_test.go

```diff
@@ -0,0 +1,468 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+import (
+	"container/list"
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"slices"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+	"golang.org/x/sync/errgroup"
+)
+
+const querierForgetDelay = 0
+const maxOutStandingPerTenant = 1000000
+
+// newBenchmarkRequestQueue mirrors the constructor for RequestQueue, but with the ability to pass in a Tree
```

Comment:
Could probably just call `q := NewRequestQueue()` in here with all the same args, then set `q.tree = tree`, right?